docs/sources/mimir/release-notes/v2.11.md

```diff
@@ -0,0 +1,92 @@
+- **Reject gRPC push requests that the ingester/distributor is unable to accept before reading them into memory.** You can enable this feature by using the `-ingester.limit-inflight-requests-using-grpc-method-limiter` and/or the `-distributor.limit-inflight-requests-using-grpc-method-limiter` CLI flags for the ingester and/or the distributor, respectively.
+- **Customize the memcached client write and read buffer size.** The buffer allocated for each memcached connection can be configured via the following CLI flags:
+  - For the blocks storage:
+    - `-blocks-storage.bucket-store.chunks-cache.memcached.read-buffer-size-bytes`
+    - `-blocks-storage.bucket-store.chunks-cache.memcached.write-buffer-size-bytes`
+    - `-blocks-storage.bucket-store.index-cache.memcached.read-buffer-size-bytes`
+    - `-blocks-storage.bucket-store.index-cache.memcached.write-buffer-size-bytes`
+    - `-blocks-storage.bucket-store.metadata-cache.memcached.read-buffer-size-bytes`
+    - `-blocks-storage.bucket-store.metadata-cache.memcached.write-buffer-size-bytes`
+  - For the query frontend:
+    - `-query-frontend.results-cache.memcached.read-buffer-size-bytes`
+    - `-query-frontend.results-cache.memcached.write-buffer-size-bytes`
+  - For the ruler storage:
+    - `-ruler-storage.cache.memcached.read-buffer-size-bytes`
+    - `-ruler-storage.cache.memcached.write-buffer-size-bytes`
+- **Configure the number of long-living workers used to process gRPC requests.** This can decrease CPU usage by reducing the number of stack allocations. Configure this feature by using the `-server.grpc.num-workers` CLI flag.
+- **Enforce a limit in bytes on the `PostingsForMatchers()` cache used by ingesters.** This limit can be configured via the `-blocks-storage.tsdb.head-postings-for-matchers-cache-max-bytes` and `-blocks-storage.tsdb.block-postings-for-matchers-cache-max-bytes` CLI flags.
+- ** Pre-allocate the pool of workers in the distributor that are used to send push requests to ingesters.** This can decrease CPU usage by reducing the number of stack allocations. You can enable this feature by using the `-distributor.reusable-ingester-push-worker` flag.
+
+### Helm chart improvements
+
+The Grafana Mimir and Grafana Enterprise Metrics Helm chart is now released independently. See the [Grafana Mimir Helm chart documentation](/docs/helm-charts/mimir-distributed/latest/).
+
+## Important changes
+
+In Grafana Mimir 2.11 the following behavior has changed:
+
+- The distributor `Push()` endpoint will now return the following gRPC codes instead of HTTP status codes:
+  - `http.StatusAccepted` (202) code is replaced with `codes.AlreadyExists`.
+  - `http.BadRequest` (400) code is replaced with `codes.FailedPrecondition`.
+  - `http.StatusTooManyRequests` (429) and the non-standard `529` (The service is overloaded) codes are replaced with `codes.ResourceExhausted`.
```

Comment:
[nit] Using the Golang constant names is likely unnecessary and possibly confusing here - I've replaced these with their numeric values and the names used in the [language-agnostic gRPC documentation](https://github.com/grpc/grpc/blob/master/doc/statuscodes.md).

```suggestion
  - 202 (accepted) code is replaced with 6 (`ALREADY_EXISTS`).
  - 400 (bad request) code is replaced with 9 (`FAILED_PRECONDITION`).
  - 429 (too many requests) and the non-standard 529 (service is overloaded) codes are replaced with 8 (`RESOURCE_EXHAUSTED`).
```